packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx

```diff
@@ -16,361 +8,341 @@ import {
+    navigationIdleDelay: 250,
+  }));
+
+  const { setTimeout, clearTimeout } = useSetTimeout();
+
+  useDebugWarnings(props);
+
+  const {
+    colorCells,
+    cellShape = 'circle',
+    columnCount,
+    // eslint-disable-next-line deprecation/deprecation
+    ariaPosInSet = props.positionInSet,
+    // eslint-disable-next-line deprecation/deprecation
+    ariaSetSize = props.setSize,
+    shouldFocusCircularNavigate = true,
+    className,
+    disabled = false,
+    doNotContainWithinFocusZone,
+    styles,
+    cellMargin = 10,
+    defaultSelectedId,
+    onChange,
+    onCellHovered,
+  } = props;
 
-  public render(): JSX.Element | null {
-    const {
-      colorCells,
-      columnCount,
-      // eslint-disable-next-line deprecation/deprecation
-      ariaPosInSet = this.props.positionInSet,
-      // eslint-disable-next-line deprecation/deprecation
-      ariaSetSize = this.props.setSize,
-      shouldFocusCircularNavigate,
-      className,
-      doNotContainWithinFocusZone,
-      styles,
-      cellMargin,
-    } = this.props;
-
-    const classNames = getClassNames(styles!, {
-      theme: this.props.theme!,
-      className,
-      cellMargin,
+  /**
+   *  Add an index to each color cells. Memoizes this so that color cells do not re-render on every update.
+   */
+  const getItemsWithIndex = React.useMemo(() => {
+    return colorCells.map((item, index) => {
+      return { ...item, index: index };
     });
+  }, [colorCells]);
 
-    if (colorCells.length < 1 || columnCount < 1) {
-      return null;
-    }
-    return (
-      <Grid
-        {...this.props}
-        id={this._id}
-        items={this._getItemsWithIndex(colorCells)}
-        columnCount={columnCount}
-        onRenderItem={this._renderOption}
-        ariaPosInSet={ariaPosInSet}
-        ariaSetSize={ariaSetSize}
-        shouldFocusCircularNavigate={shouldFocusCircularNavigate}
-        doNotContainWithinFocusZone={doNotContainWithinFocusZone}
-        onBlur={this._onSwatchColorPickerBlur}
-        theme={this.props.theme!}
-        styles={{
-          root: classNames.root,
-          tableCell: classNames.tableCell,
-          focusedContainer: classNames.focusedContainer,
-        }}
-      />
-    );
-  }
+  const [selectedId, setSelectedId] = useControllableValue(props.selectedId, defaultSelectedId, onChange);
+
+  const classNames = getClassNames(styles!, {
+    theme: props.theme!,
+    className,
+    cellMargin,
+  });
+
+  const gridStyles = {
+    root: classNames.root,
+    tableCell: classNames.tableCell,
+    focusedContainer: classNames.focusedContainer,
+  };
 
   /**
    * When the whole swatchColorPicker is blurred,
    * make sure to clear the pending focused stated
    */
-  private _onSwatchColorPickerBlur = (): void => {
-    if (this.props.onCellFocused) {
-      this._cellFocused = false;
-      this.props.onCellFocused();
+  const onSwatchColorPickerBlur = React.useCallback((): void => {
+    if (props.onCellFocused) {
```

Comment:
Add this to the initial props destructuring so you only need a dep on `onCellFocused` itself, not the whole props object